src/main/java/frc/robot/scoreStates/AmpState.java

```diff
@@ -8,49 +8,88 @@
-                    double robotRotation = botPose.getRotation().getRadians();
-                    boolean isGripperReversed =
-                            false; // TODO: replace with actual gripper.isForward
-                    double distance = Utils.getDistanceFromPoint(ampPose, botPose);
-                    boolean hasTimeToTurnGripper =
+                    robotRotation = botPose.getRotation().getRadians();
+                    distance = Utils.getDistanceFromPoint(ampPose, botPose);
+                    hasTimeToTurnGripper =
                             distance
                                     > ScoreStateConstants.MIN_DISTANCE_TO_TURN_GRIPPER.in(
                                             Units.Meters);
-
-                    if ((isGripperReversed && robotRotation < 0) || hasTimeToTurnGripper) {
+                    if ((gripper.isReversed() && robotRotation < 0) || hasTimeToTurnGripper) {
+                        isAmpingForward = false;
                         ampRotation = ScoreStateConstants.AMP_ROTATION_REVERSE;
                     }
-                    return AutoBuilder.pathfindToPose(
-                            new Pose2d(ampPose, ampRotation), Constants.AUTO_CONSTRAINTS);
-                },
+                });
+    }
+
+    @Override
+    public Command driveToClosestOptimalPoint() {
+        return Commands.defer(
+                () ->
+                        AutoBuilder.pathfindToPose(
+                                new Pose2d(ampPose, ampRotation), Constants.AUTO_CONSTRAINTS),
                 Set.of(SwerveDrive.getInstance()));
     }
 
     @Override
-    public Command stateInitialize() {
-        return null;
+    public Command initializeSubsystem() {
+        return Commands.defer(
+                () ->
+                        Commands.parallel(
+                                elevator.setHeight(ElevatorConstants.MAX_HEIGHT),
+                                gripper.setWristPosition(
+                                        isAmpingForward
+                                                ? GripperConstants.WRIST_FORWARD_AMP_POSE
+                                                        .mutableCopy()
+                                                : GripperConstants.WRIST_BACKWARDS_AMP
+                                                        .mutableCopy())),
+                Set.of(gripper, elevator));
     }
 
     @Override
     public Command score() {
-        return null;
+        return driveToClosestOptimalPoint()
+                .andThen(
+                        Commands.defer(
+                                () ->
+                                        gripper.setRollerPower(
+                                                isAmpingForward
+                                                        ? GripperConstants.AMP_POWER_NORMAL
+                                                        : GripperConstants.AMP_POWER_REVERSE),
+                                Set.of(gripper)));
```

Comment:
There are a few places where using [`Commands.either()`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/Commands.html#either(edu.wpi.first.wpilibj2.command.Command,edu.wpi.first.wpilibj2.command.Command,java.util.function.BooleanSupplier)) will be more concise -- this is one of them.

Reply:
236ba0c0c1a83736ea293a3a01da0f7e105f83eb

Reply:
Awesome!

Reply:
3621c88

Reply:
Why did you get rid of using `either()`? `either()` returns a `ConditionalCommand`, if that's not clear.

Reply:
1bc986a7abf2edabe40b32660c646d4d9c5b30f1